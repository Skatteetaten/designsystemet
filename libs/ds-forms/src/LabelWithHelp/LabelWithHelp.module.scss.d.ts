declare const classNames: {
  readonly label: 'label';
  readonly label_required: 'label_required';
  readonly srOnly: 'srOnly';
};
export = classNames;
