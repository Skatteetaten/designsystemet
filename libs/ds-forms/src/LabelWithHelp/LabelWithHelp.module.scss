@use 'libs/ds-core-utils/src/styles';

.label {
  color: var(--palette-graphite-70);
  font-size: var(--font-size-m);
  font-weight: var(--font-weight-bold);
  line-height: var(--semantic-line-height-input);
  overflow-wrap: anywhere;

  &_required::after {
    margin-left: var(--spacing-xxs);
    color: var(--semantic-danger-foreground);
    content: '*';
  }
}

.srOnly {
  @include styles.sr-only;
}
