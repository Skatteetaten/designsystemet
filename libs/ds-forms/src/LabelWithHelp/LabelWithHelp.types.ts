import { ComponentPropsWithoutRef, Ref } from 'react';

import { BaseProps } from '@skatteetaten/ds-core-utils';

import { HelpProps } from '../Help/Help.types';

type RequiredLabelHTMLAttributes = Pick<
  ComponentPropsWithoutRef<'label'>,
  'htmlFor'
>;

export interface LabelWithHelpProps
  extends RequiredLabelHTMLAttributes,
    BaseProps {
  ref?: Ref<HTMLLabelElement>;
  classNames?: {
    label?: string;
  } & HelpProps['classNames'];
  /** Ledetekst */
  children: string;
  /** Skjuler hele komponenten men er fortsatt synlig for skjermleser */
  hideLabel?: boolean;
  /** Tilleggstekst */
  description?: string;
  /** Hjelpetekst */
  helpText?: HelpProps['helpText'];
  /** Overskriver default hjelpeikon */
  helpSvgPath?: HelpProps['helpSvgPath'];
  /** Overskriver default tooltip-tekst til hjelpeikon */
  titleHelpSvg?: HelpProps['titleHelpSvg'];
  /** Om obligatorisk label skal markeres med stjerne */
  showRequiredMark?: boolean;
  /** Callback som kalles n√•r hjelpetekst vises/skjules */
  onHelpToggle?: HelpProps['onHelpToggle'];
}
