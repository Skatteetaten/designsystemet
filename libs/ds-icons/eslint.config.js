import nxPlugin from '@nx/eslint-plugin';

import baseConfig from '../../eslint.config.js';

export default [
  ...baseConfig,
  ...nxPlugin.configs['flat/react'],
  {
    ignores: ['!**/*'],
  },
  {
    files: ['**/*.ts', '**/*.tsx', '**/*.js', '**/*.jsx'],
    rules: {},
  },
  {
    files: ['**/*.ts', '**/*.tsx'],
    rules: {},
  },
  {
    files: ['**/*.js', '**/*.jsx'],
    rules: {},
  },
];
