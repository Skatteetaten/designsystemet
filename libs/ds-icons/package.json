{
  "name": "@skatteetaten/ds-icons",
  "groupId": "no.skatteetaten.aurora",
  "version": "1.6.0",
  "license": "Apache-2.0",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "type": "module",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^14",
    "@skatteetaten/ds-core-utils": "1.6.0"
  },
  "scripts": {},
  "dependencies": {},
  "devDependencies": {}
}
