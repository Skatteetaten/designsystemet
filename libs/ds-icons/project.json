{
  "name": "ds-icons",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-icons/src",
  "projectType": "library",
  "tags": [],
  "implicitDependencies": ["!ds-dev-config"],
  "targets": {
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-icons",
        "tsConfig": "libs/ds-icons/tsconfig.lib.json",
        "project": "libs/ds-icons/package.json",
        "entryFile": "libs/ds-icons/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-icons/rollup.config.cjs"],
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/ds-icons/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/ds-icons"],
      "options": {
        "jestConfig": "libs/ds-icons/jest.config.ts"
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-icons/**/*.css",
          "libs/ds-icons/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      },
      "dependsOn": [
        {
          "projects": "ds-core-designtokens",
          "target": "build"
        }
      ]
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-icons"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-icons {args.switches}"
      }
    }
  }
}
