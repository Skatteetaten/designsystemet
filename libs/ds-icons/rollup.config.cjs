const path = require('path');

const createRollupConfig = require('../ds-dev-config/build/rollup-config-factory.cjs');

const srcDir = path.resolve(__dirname, `src`);
const pathToTsConfig = 'libs/ds-icons/tsconfig.lib.json';
const pathToCSS = path.resolve(__dirname, `**/*.{css,scss}`);
const outputDir = path.resolve('./', `dist/libs/ds-icons/`);

module.exports = (config) =>
  createRollupConfig(config, srcDir, pathToTsConfig, pathToCSS, outputDir);
