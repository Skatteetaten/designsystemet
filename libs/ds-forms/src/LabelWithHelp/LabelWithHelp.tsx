import { forwardRef, JSX, useId } from 'react';

import { getCommonClassNameDefault } from '@skatteetaten/ds-core-utils';

import { LabelWithHelpProps } from './LabelWithHelp.types';
import { Help } from '../Help/Help';

import styles from './LabelWithHelp.module.scss';

export const LabelWithHelp = forwardRef<HTMLLabelElement, LabelWithHelpProps>(
  (
    {
      id: idExternal,
      className = getCommonClassNameDefault(),
      classNames,
      lang,
      'data-testid': dataTestId,
      description,
      helpText,
      helpSvgPath,
      titleHelpSvg,
      htmlFor,
      hideLabel,
      showRequiredMark,
      children,
      onHelpToggle,
    },
    ref
  ): JSX.Element => {
    const uniqueLabelId = `labelId-${useId()}`;
    const labelId = idExternal ?? uniqueLabelId;

    const requiredMarkClassName = showRequiredMark ? styles.label_required : '';
    const hideLabelClassName = hideLabel ? styles.srOnly : '';
    const concatenatedClassName = `${
      styles.label
    } ${requiredMarkClassName} ${hideLabelClassName} ${className ?? ''} ${
      classNames?.label ?? ''
    }`.trim();

    return (
      <>
        <label
          ref={ref as (instance: HTMLLabelElement | null) => void}
          id={labelId}
          data-testid={dataTestId}
          className={concatenatedClassName}
          lang={lang}
          htmlFor={htmlFor}
        >
          {children}
        </label>
        <Help
          className={classNames?.helpText}
          classNames={classNames}
          helpSvgPath={helpSvgPath}
          helpText={helpText}
          hideHelp={hideLabel}
          targetId={labelId}
          titleHelpSvg={titleHelpSvg}
          description={description}
          onHelpToggle={onHelpToggle}
        />
      </>
    );
  }
);

LabelWithHelp.displayName = 'LabelWithHelp';
