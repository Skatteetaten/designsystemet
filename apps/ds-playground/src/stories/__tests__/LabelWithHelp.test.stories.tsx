import { WarningSVGpath } from '@skatteetaten/ds-icons';
import { Meta, StoryObj } from '@storybook/react';
import { expect, fireEvent, within } from '@storybook/test';

import {
  loremIpsumWithoutSpaces,
  wrapper,
} from './testUtils/storybook.testing.utils';
// eslint-disable-next-line @nx/enforce-module-boundaries
import { LabelWithHelp } from '../../../../../libs/ds-forms/src/LabelWithHelp/LabelWithHelp';
// eslint-disable-next-line @nx/enforce-module-boundaries
import { LabelWithHelpProps } from '../../../../../libs/ds-forms/src/LabelWithHelp/LabelWithHelp.types';
import { SystemSVGPaths } from '../utils/icon.systems';

const meta = {
  component: LabelWithHelp,
  title: 'Tester/LabelWithHelp (intern)',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    classNames: { table: { disable: true } },
    children: { table: { disable: true } },
    description: { table: { disable: true } },
    helpText: { table: { disable: true } },
    helpSvgPath: {
      table: { disable: true },
      options: Object.keys(SystemSVGPaths),
      mapping: SystemSVGPaths,
    },
    hideLabel: { table: { disable: true } },
    showRequiredMark: { table: { disable: true } },
    titleHelpSvg: { table: { disable: true } },
    // HTML
    htmlFor: { table: { disable: true } },
    // Events
    onHelpToggle: { table: { disable: true } },
  },
} satisfies Meta<typeof LabelWithHelp>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultDescription = 'Kort hjelpetekst';
const defaultHelpText =
  'Vi trenger Ã¥ vite navnet ditt dersom vi skal kontakte deg senere.';
const defaultChildrenText = 'Ledetekst';
const defaultArgs: LabelWithHelpProps = {
  children: defaultChildrenText,
};

export const WithRef = {
  name: 'With Ref (FA1)',
  args: {
    ...defaultArgs,
    ref: (instance: HTMLLabelElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },
  argTypes: {
    ref: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: { disable: true },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const label = canvas.getByText(defaultChildrenText);
    await expect(label).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5)',
  args: {
    ...defaultArgs,
    id: 'htmlid',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const label = canvas.getByText(defaultChildrenText);
    await expect(label).toHaveAttribute('id', 'htmlid');
    await expect(label).toHaveClass('dummyClassname');
    await expect(label).toHaveAttribute('lang', 'nb');
    await expect(label).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (FS-A1, FS-A2, FS-A4, FS-B1)',
  args: {
    ...defaultArgs,
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const label = canvas.getByText(defaultChildrenText);
    await expect(label.tagName).toBe('LABEL');
  },
} satisfies Story;

export const WithDescription = {
  name: 'With Description (FS-A3)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
  },
  argTypes: {
    description: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const descriptionNode = canvas.getByText(defaultDescription);
    await expect(descriptionNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithShowRequiredMark = {
  name: 'With ShowRequiredMark (FS-A4)',
  args: {
    ...defaultArgs,
    showRequiredMark: true,
  },
  argTypes: {
    showRequiredMark: { table: { disable: false } },
  },
} satisfies Story;

export const WithHideLabel = {
  name: 'With HideLabel (FS-A7)',
  args: {
    ...defaultArgs,
    description: defaultDescription,
    helpText: defaultHelpText,
    hideLabel: true,
  },
  argTypes: {
    hideLabel: { table: { disable: false } },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const label = canvas.getByText(defaultChildrenText);
    const descriptionNode = canvas.getByText(defaultDescription);
    const helpButtonNode = canvas.getByRole('button');
    await expect(label).toBeInTheDocument();
    await expect(descriptionNode).toBeInTheDocument();
    await expect(helpButtonNode).toBeInTheDocument();
  },
} satisfies Story;

export const WithHelpTextSvgPathAndTitle = {
  name: 'With HelpText HelpSvgPath And TitleHelpSvg (FS-A1, FS-A5 delvis, FS-B2 delvis)',
  args: {
    ...defaultArgs,
    helpText: defaultHelpText,
    helpSvgPath: WarningSVGpath,
    titleHelpSvg: 'Tooltip',
  },
  argTypes: {
    helpText: { table: { disable: false } },
    helpSvgPath: { table: { disable: false } },
    titleHelpSvg: { table: { disable: false } },
  },
  parameters: {
    imageSnapshot: {
      click: `${wrapper} > button`,
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const helpButton = canvas.getByRole('button', {
      description: defaultChildrenText,
    });
    await expect(helpButton).toBeInTheDocument();
    const svgNode = canvas.getByLabelText('Tooltip', { selector: 'svg' });
    await expect(svgNode).toBeInTheDocument();
    const label = canvas.getByText(defaultChildrenText);
    await expect(helpButton).toHaveAttribute('aria-describedby', label.id);
  },
} satisfies Story;

export const WithLongChildren = {
  name: 'With Long Children',
  args: {
    ...defaultArgs,
    children: loremIpsumWithoutSpaces,
  },
  argTypes: {
    children: { table: { disable: false } },
  },
  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithHelpToggleEvent = {
  name: 'With onHelpToggle Event',
  args: {
    ...defaultArgs,
    helpText: 'Hjelpetekst',
    onHelpToggle: (isOpen: boolean): void => {
      alert(isOpen ? 'Hjelpetekst blir vist' : 'Hjelpetekst skjules');
    },
  },
  parameters: {
    imageSnapshot: {
      disable: true,
    },
  },
} satisfies Story;

export const WithCustomClassNames = {
  name: 'With Custom ClassNames (FA3)',
  args: {
    ...defaultArgs,
    classNames: {
      label: 'dummyClassname',
      description: 'dummyClassname',
      helpText: 'dummyClassname',
    },
    description: 'beskrivelse',
    helpText: 'hjelp',
  },
  argTypes: {
    classNames: {
      table: { disable: false },
    },
  },
  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // eslint-disable-next-line testing-library/no-node-access
    const label = canvas.getByText(defaultChildrenText);
    await expect(label).toHaveClass('dummyClassname');
    await expect(canvas.getByText('beskrivelse')).toHaveClass('dummyClassname');

    await fireEvent.click(canvas.getByRole('button'));
    const helpText = canvas.getByText('hjelp').parentElement;
    await expect(helpText).toHaveClass('dummyClassname');
  },
} satisfies Story;
