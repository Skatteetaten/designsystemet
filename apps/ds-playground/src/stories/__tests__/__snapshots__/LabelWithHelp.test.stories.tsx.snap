// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/LabelWithHelp (intern) Defaults play-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:r2:"
         class="LabelWithHelp_label__59tP-"
  >
    Ledetekst
  </label>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithAttributes play-test 1`] = `
<div data-test-block="true">
  <label id="htmlid"
         data-testid="123ID"
         class="LabelWithHelp_label__59tP-   dummyClassname"
         lang="nb"
  >
    Ledetekst
  </label>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithCustomClassNames play-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:rd:"
         class="LabelWithHelp_label__59tP-    dummyClassname"
  >
    Ledetekst
  </label>
  <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Help_helpButton__I0E-6"
          type="button"
          aria-describedby="labelId-:rd:"
          aria-expanded="true"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:re:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:re:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
  <div class="Help_helpBox__nbhjO  dummyClassname dummyClassname">
    <span class="Help_helpBoxText__j6sH7">
      hjelp
    </span>
    <button class="IconButton_button__mw9sG IconButton_button_small__etst4  Help_helpBoxCloseButton__RueX2"
            type="button"
    >
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
           viewbox="0 0 24 24"
           focusable="false"
           aria-labelledby="svgtitle-:rf:"
           aria-hidden="false"
           role="img"
      >
        <title id="svgtitle-:rf:">
          Lukk
        </title>
        <path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z">
        </path>
      </svg>
    </button>
  </div>
  <div class="Help_description__niabw Help_descriptionMarginTop__YMqoC  dummyClassname">
    beskrivelse
  </div>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithDescription play-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:r3:"
         class="LabelWithHelp_label__59tP-"
  >
    Ledetekst
  </label>
  <div class="Help_description__niabw">
    Kort hjelpetekst
  </div>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithHelpTextSvgPathAndTitle play-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:r7:"
         class="LabelWithHelp_label__59tP-"
  >
    Ledetekst
  </label>
  <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Help_helpButton__I0E-6"
          type="button"
          aria-describedby="labelId-:r7:"
          aria-expanded="false"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r8:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r8:">
        Tooltip
      </title>
      <path d="M13 14h-2v-4h2v4Zm0 4h-2v-2h2v2ZM1 21h22L12 2 1 21Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithHelpToggleEvent smoke-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:rb:"
         class="LabelWithHelp_label__59tP-"
  >
    Ledetekst
  </label>
  <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Help_helpButton__I0E-6"
          type="button"
          aria-describedby="labelId-:rb:"
          aria-expanded="false"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:rc:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:rc:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithHideLabel play-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:r5:"
         class="LabelWithHelp_label__59tP-  LabelWithHelp_srOnly__uWL7Z"
  >
    Ledetekst
  </label>
  <button class="IconButton_button__mw9sG IconButton_button_extraSmall__3+QfY IconButton_button_outlined__aDbNH Help_helpButton__I0E-6 Help_srOnly__svaKL"
          type="button"
          aria-describedby="labelId-:r5:"
          aria-expanded="false"
  >
    <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz IconButton_icon__1J4sN"
         viewbox="0 0 24 24"
         focusable="false"
         aria-labelledby="svgtitle-:r6:"
         aria-hidden="false"
         role="img"
    >
      <title id="svgtitle-:r6:">
        Hjelp
      </title>
      <path d="M10 19h3v3h-3v-3Zm2-17c5.35.22 7.68 5.62 4.5 9.67-.83 1-2.17 1.66-2.83 2.5C13 15 13 16 13 17h-3c0-1.67 0-3.08.67-4.08.66-1 2-1.59 2.83-2.25C15.92 8.43 15.32 5.26 12 5a3 3 0 0 0-3 3H6a6 6 0 0 1 6-6Z">
      </path>
    </svg>
  </button>
  <div class="Help_description__niabw  Help_srOnly__svaKL">
    Kort hjelpetekst
  </div>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithLongChildren smoke-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:ra:"
         class="LabelWithHelp_label__59tP-"
  >
    Loremipsumdolorsitamet.Allesomharlagetennettside,trengtlittfylltekstellerbaresurfetrundtp√•nettetharantageligvissettdisseordene,etterfulgtaventilsynelatendeeviglangtekstfyltmedlatinskeliksomsetninger.
  </label>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithRef play-test 1`] = `
<div data-test-block="true">
  <label id="dummyIdForwardedFromRef"
         class="LabelWithHelp_label__59tP-"
  >
    Ledetekst
  </label>
</div>
`;

exports[`Tester/LabelWithHelp (intern) WithShowRequiredMark smoke-test 1`] = `
<div data-test-block="true">
  <label id="labelId-:r4:"
         class="LabelWithHelp_label__59tP- LabelWithHelp_label_required__KSuzW"
  >
    Ledetekst
  </label>
</div>
`;
